@here/harp-mapview/lib/MapView.ts

```diff
@@ -1839,6 +1844,32 @@ export class MapView extends THREE.EventDispatcher {
         this.m_visibleTiles.markTilesDirty(dataSource);
     }
 
+    /**
+     * Sets the DataSource which contains the elevations, the elevation range source, and the
+     * elevation provider. Only a single elevation source is possible per [[MapView]]
+     *
+     * If the terrain-datasource is merged with this repository, we could internally construct
+     * the [[ElevationRangeSource]] and the [[ElevationProvider]] and access would be granted to
+     * the application when it asks for it, to simplify the API.
+     *
+     * @param elevationSource The datasource containing the terrain tiles.
+     * @param elevationRangeSource Allows access to the elevation min / max per tile.
+     * @param elevationProvider Allows access to the elevation at a given location or a ray
+     *      from the camera.
+     */
+    setElevationSource(
+        elevationSource: DataSource,
```

Comment:
Will we ever have an ElevationProvider that is not associated with a DataSource?
If not you could move the DataSource inside the ElevationProvider interface.
You could also drop the getTilingScheme() method as it should be equal to getDataSource().getTilingScheme().

Reply:
Good question, the ElevationProvider uses the in memory terrain mesh and applies the functions raycast / getHeight to that, so it indirectly needs it, but there isn't a tight coupling (see the TerrainElevationProvider on gerrit). My feeling is that it isn't so necassary to combine, but let's discuss in the office and maybe there is something I'm missing.

Reply:
@CraigBerlin: We are not 100% sure how we want to render the terrain if we have vector data. In harp-native we still rendered the terrain mesh and then the vector data on top, but instead we could also just render the elevated vector data (given it covers the whole tile) so the actual terrain data source would not be used.